core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -61,7 +60,7 @@ private[spark] class UnionPartition[T: ClassTag](
 
 object UnionRDD {
   private[spark] lazy val partitionEvalTaskSupport =
-    new ForkJoinTaskSupport(new ForkJoinPool(8))
+    new ForkJoinTaskSupport(ThreadUtils.newForkJoinPool("partition-eval-task-support", 8))
```

Comment:
This is from review comment in #24113 which makes sense and also helps to centralize dealing with deprecated Scala ForkJoinPool.